rpcauth/backends.py

```diff
@@ -0,0 +1,65 @@
+# Copyright The IETF Trust 2023, All Rights Reserved
+# -*- coding: utf-8 -*-
+
+from django.core.exceptions import SuspiciousOperation
+from django.db import IntegrityError
+from mozilla_django_oidc.auth import OIDCAuthenticationBackend
+
+from rpc.models import DatatrackerPerson
+
+
+class RpcOIDCAuthBackend(OIDCAuthenticationBackend):
+    """Customized OIDC auth backend
+
+    Assumes the datatracker makes the following claims:
+      sub - pk of the Person who was authenticated, as a string
+      name - plain_name of the Person identified by sub
+      roles - list of 2-tuples corresponding to Person's active roles
+    """
+
+    def create_user(self, claims):
+        """Create a User following a successful auth"""
+        subject_id = claims["sub"]
+        # A DatatrackerPerson may exist without a user so use get_or_create()
+        datatracker_person, _ = DatatrackerPerson.objects.get_or_create(
+            subject_id=subject_id,
+            plain_name=claims.get("name", ""),
+        )
+        try:
+            # Rely on the OneToOne field to avoid creating duplicate Users for a DatatrackerPerson
+            new_user = self.UserModel.objects.create(
+                username=f"dt-user-{subject_id}",
```

Comment:
Consider dt-person, so future us aren't nudged into looking in the datatracker user tables.